### PR TITLE
doc: adds missing CONTRIBUTING.md and related templates. closes #78

### LINKED ISSUE #78: Add contributing guidelines
After checking more carefully the repo, I have noticed given the lack of conventions, there is considerable heterogeneity in the repo. To help keeping the repo in a good shape, I think we should add a specific CONTRIBUTING.md file for this repo (I volunteer to do it).

I would like to include things like:
* I want to contribute, what should I work on?
* Before filling and issue
* Getting a development environment set up
* Adding a new expression
* Commit message format
* Pull requests

So, I'm happy to prepare one, but I would like to get your thoughts @ekenes, @MikeMillerGIS, @VwakeM, @ted-howard, @jdcarls2, about some conventions I think we should define, specially about `Adding a new expression`.

My questions for you are:

1. Should we specify a convention for file & folder names capitalization? ([camelCase](https://en.wikipedia.org/wiki/Camel_case) vs [snake_case](https://en.wikipedia.org/wiki/Snake_case) vs [PascalCase](https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference/#pascal-case) vs or [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case)) <- Any preference? 
	1.1. Should README.md always capitalized? ([example](https://github.com/Esri/arcade-expressions/blob/master/dictionary_renderer/Conduit/Readme.md))
	1.2. Is it OK to use special characters like "(", ")", ... ([example](https://github.com/Esri/arcade-expressions/tree/master/dashboard_data))
2. If an expression is too long, is it recommended keeping it in a *.js file instead of the markdown file? ([example](https://github.com/Esri/arcade-expressions/blob/master/attribute_rule_calculation/CreateLateralDevSummitPlenary2023.md))
3. Should we add images and sample data within folders? (e.g. `images/` & `sample-data/`). I have seen some folders like [attribute_rule_calculation](https://github.com/Esri/arcade-expressions/tree/master/attribute_rule_calculation) with ton of files.
4. I sometimes see we have folders grouping expressions ([example](https://github.com/Esri/arcade-expressions/tree/master/any/barcode-gas)). Should they have their own README.md and list the expressions there instead of in the parent README file? ([Barcode example](https://github.com/Esri/arcade-expressions/tree/master/any))
5. Should we request a local copy of the data as a static file? https://gdb.zip/
  5.1. If so, should we use specific extensions? (e.g. [gdb.zip](https://github.com/Esri/arcade-expressions/blob/master/attribute_rule_calculation/LastValue.gdb.zip) vs [zip](https://github.com/Esri/arcade-expressions/blob/master/attribute_rule_calculation/CreateLateralDevSummitPlenary2023.zip) vs ...; [stylx](https://github.com/Esri/arcade-expressions/tree/master/dictionary_renderer/Conduit))
6. Do we need to add. `Contributing` and `License` to all README.md files?

My proposal:

1. I would vote for kebab-case
	1.1. Yes, I would recommend README.md (capitals) 
	1.2. I would say avoid `\/:*?"<>|`
2. I'm not against that, but if so, I would place it in a `expressions` folder with the same name as the markdown file
3. I would say yes, use folders.
4. I would say yes. Add a single link in the parent README.md to the child README.md and list the expressions there
5. I would say it is recommended whenever possible, and set `[extension].zip` meaning ____.gdb.zip, ____.stylx.zip, ...
6. I would say no.

Thanks in advance! 😊

P.S. sorry if I added or missed someone here, please let me know if that's the case.

### PR DESCRIPTION
**Pull request by hhkaos**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+* If you have a **general question about Arcade**, please check the [arcade tag at Esri Community](https://community.esri.com/t5/tag/arcade/tg-p), or go to the [right place](https://github.com/Esri/arcade-expressions/issues/new/choose) where you can ask questions, and collaborate with others.
+  * If you think Arcade should have its own community, you might want to consider [giving kudos to this idea](https://community.esri.com/t5/community-ideas/arcade-community/idi-p/1131578).
+* Provide all of the requested info from the appropriate [issue template](https://github.com/Esri/arcade-expressions/issues/new/choose). 
+* **Other things to consider when filling a bug issue**(so we can work on resolving the issue as soon as possible):
+  - Use a clear and descriptive title.
+  - Detail what is happening now vs what should happen.
+  - Tell us how to reproduce the issue (e.g. is it happening in a specific profile, browser, ...?).
+  - Can it be reliably reproduced? If not, tell us how often it happens and under what circumstances.
+  - Provide a sample that reproduces the issue  whenever possible.
+  - Screenshots, GIFsn and screen recordings are our friends!.
```
Comment (ekenes) [inline CONTRIBUTING.md:43]:
```suggestion
  - Screenshots, GIFs, and screen recordings are our friends!
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  - Did this problem start happening after a recent release or was it always a bug?.
```
Comment (ekenes) [inline CONTRIBUTING.md:44]:
```suggestion
  - Did this problem start happening after a recent release or was it always a bug?
```

**Review by ekenes — APPROVED** [review]
Looks good! Just some minor typos that need cleaning up, then I'll merge.

**Comment by hhkaos** [comment]
@ekenes thanks, I think I have fixed the typos 😄

**Pull request merged**